paper/src/main/java/com/denizenscript/denizen/paper/properties/PaperEntityExtensions.java

```diff
@@ -184,6 +186,40 @@ public static void register() {
                 }
                 object.getLivingEntity().damageItemStack(slot.asEnum(EquipmentSlot.class), amount.asInt());
             });
+
+            // <--[mechanism]
+            // @object EntityTag
+            // @name shear
+            // @Plugin paper
+            // @group paper
+            // @description
+            // Shears a sheep, harvests a bogged or mushroom cow or derps a snow golem as a player would do, including drops.
+            // If the entity is not ready to be shared, there are no drops but the sound will still play.
+            // Optionally, specify a sound source to change the source of the sound.
+            // Valid types are AMBIENT, BLOCKS, HOSTILE, MASTER, MUSIC, NEUTRAL, PLAYERS, RECORDS, VOICE and WEATHER

```

Comment:
The JD mentions entities that change entirely when sheared (e.g. mushroom cows becoming normal cows) becoming a different entity; probably a good idea to have a note for this here (and maybe link to https://meta.denizenscript.com/Docs/Events#entity%20transforms) as in some cases you'd have to get a new `EntityTag` if you want to do more stuff after shearing, or specifically have shearing be the last thing done.